update several document related to privilege sql statement (#2828)
pingcap/docs: sql-statements/sql-statement-grant-privileges.md

# `GRANT <privileges>`

This statement allocates privileges to a pre-existing user in TiDB. The privilege system in TiDB follows MySQL, where credentials are assigned based on a database/table pattern.

> Executing this statement require `GRANT OPTION` privilege and all peivileges you want to allocate.

Reviewer: ```suggestion
Executing this statement requires the `GRANT OPTION` privilege and all privileges you allocate.
```
